Lib/code.py

```diff
@@ -150,10 +150,21 @@ def showtraceback(self, **kwargs):
             else:
                 # If someone has set sys.excepthook, we let that take precedence
                 # over self.write
-                sys.excepthook(ei[0], ei[1], last_tb)
+                self._call_excepthook(ei[0], ei[1], last_tb)
```

Comment:
Should `lines` even be computed if `sys.excepthook` is not `sys.__excepthook__`?

What will happen if `traceback.format_exception()` or `self.write()` raise an exception?

Reply:
Re `lines`: this has always been done that way, but since I am touching the code it indeed makes sense to fix that too (the `lines` computation is also done in the equivalent `if` in `showsyntaxerror` for example). done.